public/app/features/plugins/extensions/usePluginExtensions.tsx

```diff
@@ -1,50 +1,41 @@
+import { useMemo } from 'react';
 import { useObservable } from 'react-use';
 
 import { PluginExtension } from '@grafana/data';
-import { GetPluginExtensionsOptions, UsePluginExtensionsResult } from '@grafana/runtime';
+import { GetPluginExtensionsOptions, useLocationService, UsePluginExtensionsResult } from '@grafana/runtime';
 
-import { getPluginExtensions } from './getPluginExtensions';
+import { useDispatch, useSelector } from '../../../types';
+
+import { getPluginExtensions, getSidecarHelpers } from './getPluginExtensions';
 import { ReactivePluginExtensionsRegistry } from './reactivePluginExtensionRegistry';
 
 export function createUsePluginExtensions(extensionsRegistry: ReactivePluginExtensionsRegistry) {
   const observableRegistry = extensionsRegistry.asObservable();
```

Comment:
The cache is removed and instead the getPluginExtensions is memoized. Not really sure if there was some other problem that the cache was solving. The performance between having a single global cache vs per component memoization should not be great IMHO while having single global cache can bring other problems (as the singleton pattern already does in other places).

Reply:
This is something that we wanted to fix for some time now, thanks. (We have [tests in place](https://github.com/grafana/grafana/blob/ebe42e1ada3258328a082e45e93b9d8418a73ada/public/app/features/plugins/extensions/usePluginExtensions.test.tsx#L108) to check if the hook is working correctly, if they are passing we should be fine from this perspective.)